projects/telemetry/inc/nmea.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+// Computes the checksum for a given NMEA message.
+//
+// Starts at the second characer of `message`,
+// and continues until the `*` character, or until `message_len` characters
+// are processed, whichever comes first.
+//
+// `out` must be a pointer to at least 3 `char`s.
+void nmea_compute_checksum(char *message, size_t message_len, char *out);
+
+// Computes the checksum for a given NMEA message, and compares the computed
+// checksum against the checksum in the message.
+//
+// Returns false if the given message does not have a checksum included.
+bool nmea_compare_checksum(char *message, size_t message_len);
```

Comment:
One thing about the `message_len` parameters - Usually, these length args refer to the number of bytes in the array, not the string length. Is that your intention as well?

Reply:
Please make that clear. Note that none of our other modules use that (including UART), so I would suggest switching it for consistency.

Reply:
Okay, I'll switch to number of bytes, and add that to the comment.